cpp/src/groupby/hash/aggregation_requests.cpp

```diff
@@ -57,13 +58,15 @@ std::vector<AggRequestType> compound_to_simple(
       });
 
   // Create minimal set of columns and simple operators
-  std::vector<AggRequestType> simple_requests;
+  std::vector<SimpleAggRequestCounter> simple_requests;
   for (auto& p : columns_to_ops) {
     auto col = p.first;
-    std::set<operators>& ops = p.second;
+    std::multiset<operators>& ops = p.second;
     while (not ops.empty()) {
-      simple_requests.emplace_back(col, *ops.begin());
-      ops.erase(ops.begin());
+      auto op = *ops.begin();
+      simple_requests.emplace_back(std::make_pair(col, op),
+                                   static_cast<gdf_size_type>(ops.count(op)));
+      ops.erase(op);
```

Comment:
This logic will end up with duplicates in `simple_requests` and end up redundantly computing the same aggregation during the groupby computation. Instead, it would be more efficient to compute the aggregation _once_, and then simply make copies of the result to satisfy the original requests. 

This is because by using a `multiset`, it will end up with duplicate entries for the same key, and `ops.erase(op)` erases a _single_ instance of that key, rather than all occurrences of that key.  `emplace_back` will then end up being called multiple times for the same request. 

This should be changed to either:
1. Use a `std::set` and increase the count when inserting
2. Erase all of the matching keys from the `multiset`

Reply:
From https://en.cppreference.com/w/cpp/container/multiset/erase
> size_type erase( const key_type& key ); (3)
> 3) Removes all elements with the key equivalent to key.

Reply:
Oh, duh. I confused (1) and (3) :)